graylog2-web-interface/src/views/components/common/ScrollToHint.tsx

```diff
@@ -15,23 +15,94 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
 import * as React from 'react';
-import { useEffect, useRef } from 'react';
+import { useEffect, useRef, useState } from 'react';
+import styled, { css } from 'styled-components';
+import chroma from 'chroma-js';
 
-import UIUtils from 'util/UIUtils';
+import Icon from 'components/common/Icon';
+
+const HINT_VISIBILITY_DURATION_MS = 2000;
+const HINT_WIDTH_PX = 200;
+
+const ScrollHint = styled.button(
+  ({ theme }) => css`
+    position: fixed;
+    left: calc(50% - ${HINT_WIDTH_PX / 2}px);
+    top: 50px;
+    color: ${theme.utils.readableColor(chroma(theme.colors.brand.tertiary).alpha(0.8).css())};
+    font-size: ${theme.fonts.size.huge};
+    padding: 20px;
+    z-index: 2000;
+    width: ${HINT_WIDTH_PX}px;
+    border-radius: 10px;
+    border: 0;
+    background: ${chroma(theme.colors.brand.tertiary).alpha(0.8).css()};
+  `,
+);
+
+const isElementVisibleInContainer = (target: HTMLElement, scrollContainer: HTMLElement) => {
+  const containerRect = scrollContainer.getBoundingClientRect();
+  const targetRect = target.getBoundingClientRect();
+
+  return targetRect.top >= containerRect.top && targetRect.bottom <= containerRect.bottom;
+};
 
 type Props = {
-  children: React.ReactNode;
-  value: any;
+  scrollContainer: React.RefObject<HTMLDivElement>;
+  // When the dependency changes, the hint will be displayed if this component is not visible.
+  triggerDependency: unknown;
```

Comment:
I am not sure about the prop name, it is hard to understand. Maybe something like `ifValueChanges`?